weatherman.py

```diff
@@ -0,0 +1,194 @@
+    if mean_iteration > 0:
+        average_mean_iteration = mean_humidity / mean_iteration
+        average_highest_temperature /= mean_iteration
+        average_lowest_temperature /= mean_iteration
+
+        print(f"Average Lowest Temperature: {average_lowest_temperature:.3f}")
+        print(f"Average Highest Temperature: {average_highest_temperature:.3f}")
+        print(f"Average Mean Humidity: {average_mean_iteration:.3f}%")
+    else:
+        print("No valid data found for the given month.")
+
+
+def generate_weather_bar_chart(weather_details):
+    for row in weather_details:
+        date = row['PKT']
+
+        try:
+            maximum_temperature = int(row['Max TemperatureC'])
+            minimum_temperature = int(row['Min TemperatureC'])
+            reset_code = Style.RESET_ALL
+
+            color_code = Fore.RED
+            bar = "+" * maximum_temperature
+            print(f"{color_code}{date} {bar:<25} {maximum_temperature}C{reset_code}")
+
+            color_code = Fore.BLUE
+            bar = "+" * minimum_temperature
+            print(f"{color_code}{date} {bar:<25} {minimum_temperature}C{reset_code}")
+
+        except ValueError:
+            continue 
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="Process weatherman data files.")
+    parser.add_argument(
+        "-e", "--weather_year",
+        help="For a given year, it will display the highest temperature and day,"
+             " lowest temperature and day, most humid day and humidity."
+    )
+    parser.add_argument(
+        "-a", "--weather_year_month",
+        help="For a given month, it will display the average highest temperature,"
+             " average lowest temperature, average mean humidity."
+    )
+    parser.add_argument(
+        "-c", "--weather_bar_chart",
+        help="For a given month, it will draw two horizontal bar charts on the console"
+             " for the highest and lowest temperature on each day. Highest in"
+             " red and lowest in blue."
+    )
+    return parser.parse_args()
+
+def main():
+    args = parse_arguments()
+    weather_year = None
+    weather_month = None
+
+    if args.weather_year_month:
+        weather_year, weather_month_input = args.weather_year_month.split('/')
+        try:
+            weather_month = convert_month(weather_month_input)
+        except ValueError as e:
+            print(e)
+            return
+
+    if args.weather_year:
+        weather_year = args.weather_year
+
+    if args.weather_bar_chart:
+        weather_year, weather_month_input = args.weather_bar_chart.split('/')
+        try:
+            weather_month = convert_month(weather_month_input)
+        except ValueError as e:
+            print(e)
+            return
+
+    if not weather_year:
+        print("Please provide a year using the '-e', '-a', or '-c' option.")
+    else:
+        weather_details = read_files(weather_year, weather_month)
+
+        if args.weather_year:
```

Comment:
Why are we rechecking the condition here? Can't we consolidate all the functionalities related to `args.weather_year` within a single if block?